pytest_fixtures/component/repository.py

```diff
@@ -91,6 +91,30 @@ def repo_setup():
     yield details
 
 
+@pytest.fixture(scope='module')
+def setup_content(module_org):
+    """This fixture is used to setup an activation key with a custom product attached. Used for
+    registering a host
+    """
+    org = module_org
+    custom_repo = entities.Repository(
+        product=entities.Product(organization=org).create(),
+    ).create()
+    custom_repo.sync()
+    lce = entities.LifecycleEnvironment(organization=org).create()
+    cv = entities.ContentView(
+        organization=org,
+        repository=[custom_repo.id],
+    ).create()
```

Comment:
non-blocking:
ID should not be needed since `repository` parameter is defined  in `entities.py` in a way that  `OneToMany` relation contains objects of  `Repository` type:
```python
'repository': entity_fields.OneToManyField(Repository),
```
https://github.com/SatelliteQE/nailgun/blob/46e937b6900985085155e85324c06f96b5ce9115/nailgun/entities.py#L2664

```suggestion
    cv = entities.ContentView(
        organization=org,
        repository=[custom_repo],
    ).create()
```